### PR TITLE
feat(staking): hydrate validator dropdown with all options

### PR DESCRIPTION
**Pull request by moo-onthelawn**
Requirements:
- fetch all validators and filter out any jailed or unbonded validators
- default selection of validator should still be from the preconfigured list in env.json (based on lowest stake weight)
- validators should otherwise be displayed in order of ascending commission and descending stake weight

Added some polishes
- fixed caret rendering before link out icon 

| Dropdown Open | Dropdown Closed |
| ---- | ---- | 
|Screenshot 2024-06-19 at 7 10 38 PM | Screenshot 2024-06-19 at 7 10 49 PM |

---

## Components

* `StakingPanel`  
  * Updated to not block rendering staking button on whether `selectedValidator` is set; the stake form won't let you submit without a `selectedValidator` anyways

* `ValidatorDropdown`
  * Update to set selected validator on click of the row 

## Hooks

* `hooks/useStakingValidator`
  * Updated to not block query on whether `validatorWhitelist` is a non-empty list given that we now fall back to entire validator list if it is
  * Add state management of `selectedValidator` here and do sorting on validator list here

### CODE REVIEW

File: src/hooks/useStakingValidator.ts [hunk 1]
```diff
@@ -42,9 +49,54 @@ export const useStakingValidator = () => {
 
     const response = await getValidators();
 
-    const filteredValidators = response?.validators.filter((validator) =>
+    // Filter out jailed and unbonded validators
+    const availableValidators =
+      response?.validators.filter(
+        (validator: Validator) => validator.status === 3 && validator.jailed === false
```
Comment (yogurtandjam) [inline src/hooks/useStakingValidator.ts:55]: [nit] extract 3 to a const somewhere `UNBONDED_VALIDATOR_STATUS`

File: src/hooks/useStakingValidator.ts [hunk 2]
```diff
@@ -42,9 +49,54 @@ export const useStakingValidator = () => {
...
+          MustBigNumber(validatorB.delegatorShares)
+        )
+          ? -1
+          : 1;
+      }
+      return 0;
+    };
+
+    availableValidators.sort(sortByCommission);
+    availableValidators.sort(sortByCommissionAndStakeWeight);
```
Comment (yogurtandjam) [inline src/hooks/useStakingValidator.ts:85]: i believe we can achieve this by sorting by stakeweight desc and then sorting by commission asc after, if that feels easier to read for you

File: src/hooks/useStakingValidator.ts [hunk 3]
```diff
@@ -55,31 +107,28 @@ export const useStakingValidator = () => {
...
-
-    // Find the validator with the fewest tokens
-    const validatorWithFewestTokens = filteredValidators.reduce((prev, curr) => {
-      return BigInt(curr.tokens) < BigInt(prev.tokens) ? curr : prev;
-    });
-
     return {
-      selectedValidator: validatorWithFewestTokens,
-      availableValidators: filteredValidators, // TODO: OTE-434
+      validatorWithFewestTokens,
```
Comment (yogurtandjam) [inline src/hooks/useStakingValidator.ts:111]: [q] do we still use this? 👀

**Review by yogurtandjam — APPROVED** [review]

File: src/hooks/useStakingValidator.ts (hunk 1 above)
Comment (rosepuppy) [reply src/hooks/useStakingValidator.ts:55]: You can import:
`import { BondStatus } from '@dydxprotocol/v4-client-js/build/node_modules/@dydxprotocol/v4-proto/src/codegen/cosmos/staking/v1beta1/staking';
`
and then do:
`BondStatus.BOND_STATUS_BONDED`

File: src/hooks/useStakingValidator.ts (hunk 2 above)
Comment (rosepuppy) [reply src/hooks/useStakingValidator.ts:85]: we can potentially combine the sorting into a single sort function too I think, but maybe thats a bit more confusing.

**Review by rosepuppy — APPROVED** [review]

File: src/hooks/useStakingValidator.ts (hunk 3 above)
Comment (moo-onthelawn) [reply src/hooks/useStakingValidator.ts:111]: ah ur right - unnecessary - will remove!

File: src/hooks/useStakingValidator.ts (hunk 1 above)
Comment (moo-onthelawn) [reply src/hooks/useStakingValidator.ts:55]: ah good point, should probably just be importing from `cosmos/staking/v1beta1/staking`

**Pull request merged**